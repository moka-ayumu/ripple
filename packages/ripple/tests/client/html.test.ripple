import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { mount, flushSync, track } from 'ripple';

describe('html directive', () => {
	let container;

	function render(component) {
		mount(component, {
			target: container
		});
	}

	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

  it('renders static html', () => {
    component App() {
      let str = '<div>Test</div>';

      {html str}
    }

    render(App);
    expect(container).toMatchSnapshot();
  });

  it('renders dynamic html', () => {
    component App() {
      let str = track('<div>Test</div>');

      {html @str}

      <button onClick={() => { @str = '<div>Updated</div>'; }}>{'Update'}</button>
    }

    render(App);
    expect(container).toMatchSnapshot();

    const button = container.querySelector('button');
    button.click();
    flushSync();

    expect(container).toMatchSnapshot();
  });
});