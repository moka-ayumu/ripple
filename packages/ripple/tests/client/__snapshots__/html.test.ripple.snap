// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`html directive > renders dynamic html 1`] = `
<div>
  <!---->
  <div>
    Test
  </div>
  <!---->
  <button>
    Update
  </button>
  
</div>
`;

exports[`html directive > renders dynamic html 2`] = `
<div>
  <!---->
  <div>
    Updated
  </div>
  <!---->
  <button>
    Update
  </button>
  
</div>
`;

exports[`html directive > renders static html 1`] = `
<div>
  <!---->
  <div>
    Test
  </div>
  <!---->
  
</div>
`;
